### PR TITLE
Add language changes to the release note

### PR DESCRIPTION
**Pull request by MaryamZi**
## Purpose
$title.

### CODE REVIEW

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 1]
```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
...
+##### Use `xml` as the contextually-expected type for interpolations in XML template expressions
+
+The contextually-expected type for interpolations in XML template expressions has been changed to `xml`. This ensures that a query expression as an interpolation works as expected. However, it is not an error for the static type to not be a subtype of the contextually-expected type.
+
+```ballerina
+import ballerina/io;
+
+public function main() {
+    var x1 = xml `<doc>${from int i in 0 ..< 2
+                            select xml `<num>${i}</num>`}</doc>`;
```
Comment (gimantha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:199]:
```suggestion
        select xml `<num>${i}</num>`}</doc>`;
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 2]
```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
...
+	[string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];
+
+	map<int>|error mapA = map from var element in arr
+                        	select [element[0], element[1]];
+	io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}
+
+	map<int>|error mapB = map from var element in arr
+                        	select [element[0], element[1]]
+                        	on conflict error("Duplicate Key");
+	io:println(mapB); // error("Duplicate Key")
```
Comment (gimantha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:116]:
```suggestion
    [string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];

    map<int>|error mapA = map from var element in arr
        select [element[0], element[1]];
    io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}

    map<int>|error mapB = map from var element in arr
        select [element[0], element[1]] on conflict error("Duplicate Key");
    io:println(mapB); // error("Duplicate Key")
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 3]
```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
...
+	map<int> & readonly|error mp = map from var element in arr
+                                    	select [element[0], element[1]];
+	io:println(isImmutable(mp)); // true
+
+	table<T2> & readonly tbl = table key(id) from var item in [[1, "John"], [2, "Jane"]]
+    	select {
+        	id: item[0],
+        	name: item[1]
+    	};
+	io:println(isImmutable(tbl)); // true
```
Comment (gimantha) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:245]:
```suggestion
type T2 record {
    int id;
    string name;
};

public function main() {
    T1 t = {arr: from var s in ["A", "C"]
        select s};
    io:println(t); // {"arr":["A","C"]}
    io:println(isImmutable(t)); // true


    var arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3]];
    map<int> & readonly|error mp = map from var element in arr
        select [element[0], element[1]];
    io:println(isImmutable(mp)); // true

    table<T2> & readonly tbl = table key(id) from var item in [[1, "John"], [2, "Jane"]]
        select {
            id: item[0],
            name: item[1]
        };
    io:println(isImmutable(tbl)); // true
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md [hunk 4]
```diff
@@ -68,18 +68,400 @@ A few backward-incompatible changes have been introduced during the Swan Lake Be
...
+    }
+}
+
+public function main() {
+    io:println(f3 is isolated function (int[]) returns int[]); // true, f3 is inferred to be isolated
+    io:println(f4 is isolated function ()); // true, f4 is inferred to be isolated
+}
+```
+
 ##### Compiler API
```
Comment (hasithaa) [inline release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:184]: Compiler API is not part of the Language Updates. it should be in a separate section.

Comment (MaryamZi) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:184]: Yeah, raised this already in https://github.com/ballerina-platform/ballerina-release/pull/2090#discussion_r945949385. cc @sanjana

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md (hunk 2 above)
Comment (MaryamZi) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:116]: Will change to 

```suggestion
    [string, int][] arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3], ["A", 4]];

    map<int>|error mapA = map from var element in arr
        					select [element[0], element[1]];
    io:println(mapA); // {"A":4,"B":1,"C":2,"D":3}

    map<int>|error mapB = map from var element in arr
							select [element[0], element[1]]
							on conflict error("Duplicate Key");
    io:println(mapB); // error("Duplicate Key")
```

to keep the rest of the query expression indented a bit more.

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md (hunk 1 above)
Comment (MaryamZi) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:199]: Isn't it clearer when indented from the start of the query?

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md (hunk 3 above)
Comment (MaryamZi) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:245]:
```suggestion
type T2 record {
    int id;
    string name;
};

public function main() {
    T1 t = {arr: from var s in ["A", "C"] select s};
    io:println(t); // {"arr":["A","C"]}
    io:println(isImmutable(t)); // true

    var arr = [["A", 0], ["B", 1], ["C", 2], ["D", 3]];
    map<int> & readonly|error mp = map from var element in arr
        							select [element[0], element[1]];
    io:println(isImmutable(mp)); // true

    table<T2> & readonly tbl = table key(id) from var item in [[1, "John"], [2, "Jane"]]
								select {
									id: item[0],
									name: item[1]
								};
    io:println(isImmutable(tbl)); // true
```

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md (hunk 4 above)
Comment (sanjana) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:184]: Will be fixed in #2094

**Review by gimantha — APPROVED** [review]

**Review by hasithaa — APPROVED** [review]

File: release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md (hunk 1 above)
Comment (gimantha) [reply release-notes/swan-lake-release-notes/swan-lake-2201.2.0-release-note.md:195]: indentation was a bit off from the next select clause

**Review by sanjana — APPROVED** [review]

**Pull request merged**